src/design_docs/org/lib/app.coffee

```diff
@@ -1,22 +1,38 @@
 _ = require('./underscore')
 h = require('./helpers')
-validation = require('./validation/validate')
+validate = require('./validation/validate')
 actions = require('./actions')
 audit = require('./shared/audit')
 
-dd =
-  views: {}
+auth = validate.auth
 
+dd =
+  views:
+    by_role:
+      map: (doc) ->
+        validate = require('views/lib/validation/validate')
+        if not validate._is_team(doc)
+          return
+        team_id = doc._id.slice(5)
+        for role_name, role_data of doc.roles
+          for user_id in (role_data.members or [])
+            emit([user_id, role_name, team_id])
   lists:
     get_teams: (header, req) ->
       out = []
       while(row = getRow())
         doc = row.doc
-        continue if not validation._is_team(doc)
+        continue if not validate._is_team(doc)
         team = h.add_team_perms(doc, req.userCtx)
```

Comment:
not yet. all it does is run the auth methods against the team with the logged in user and annotates the team output to include whether the user can modify each of the different components of the team.